warehouse/slave/worker.go

```diff
@@ -636,45 +636,55 @@ func (w *worker) destinationFromSlaveConnectionMap(destinationId, sourceId strin
 	return conn, nil
 }
 
-// HandleSchemaChange checks if the existing column type is compatible with the new column type
-func HandleSchemaChange(existingDataType, currentDataType model.SchemaType, value any) (any, error) {
+// HandleSchemaChange checks if the existing data type (from warehouse schema) is compatible with the inferred data type (from event data)
+func HandleSchemaChange(existingDataType, inferredDataType model.SchemaType, value any) (any, error) {
 	var (
 		newColumnVal any
 		err          error
 	)
 
 	if existingDataType == model.StringDataType || existingDataType == model.TextDataType {
 		// only stringify if the previous type is non-string/text/json
-		if currentDataType != model.StringDataType && currentDataType != model.TextDataType && currentDataType != model.JSONDataType {
+		if inferredDataType != model.StringDataType && inferredDataType != model.TextDataType && inferredDataType != model.JSONDataType {
 			newColumnVal = fmt.Sprintf("%v", value)
 		} else {
 			newColumnVal = value
 		}
-	} else if (currentDataType == model.IntDataType || currentDataType == model.BigIntDataType) && existingDataType == model.FloatDataType {
-		intVal, ok := value.(int)
-		if !ok {
-			err = fmt.Errorf("incompatible schema conversion from %v to %v", existingDataType, currentDataType)
-		} else {
-			newColumnVal = float64(intVal)
+	} else if (inferredDataType == model.IntDataType || inferredDataType == model.BigIntDataType) && existingDataType == model.FloatDataType {
+		// All warehouse destinations currently support int to float coercion.
+		// value should be float (go json unmarshals all numbers as float64), unless explicitly converted to int. Keeping logic agnostic in this function.
+		switch v := value.(type) {
+		case int:
+			newColumnVal = float64(v)
+		case int32:
+			newColumnVal = float64(v)
+		case int64:
+			newColumnVal = float64(v)
+		case float32:
+			newColumnVal = float64(v)
+		case float64:
+			newColumnVal = v
+		default:
```

Comment:
Lets add a log for the default case so that we know that the code is not going there.

Reply:
Adding log will be some decent change, need to pass logger. Any other ideas?

Reply:
No other ideas. Slave caller already has a logger. For snowpipe, the logger is present in caller's caller.